src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -363,14 +356,27 @@ class CDCL(val clauses: MutableList<MutableList<Int>>, initNumber: Int = 0) {
                     }
                 }
             }
-            addVariable(clause, lit)
+            setVariableValues(clause, lit)
```

Comment:
Hmmm... This should be named "enqueue" or "assign" to match the similar names in the wild (in other solvers). Also, change the argument order: you "assign" a literal `lit` (implicitly, you say "lit=true") due to some "reason clause".
So, consider `fun assign(lit: Lit, reason: Clause)`.